src/nebulr-auth/auth-guard.ts

```diff
@@ -45,54 +67,70 @@ export class AuthGuard implements CanActivate {
-    if (NebulrAuthService.isAnonymousUser(authResponse.user)) {
-      this._debugger.log(`canActivate no need to do plan restriction since this is a ANONYMOUS user`);
-      return true;
+    const acessRawToken = parsedRequest.request.get('Authorization') || parsedRequest.request.get('authorization');
+    if (acessRawToken && acessRawToken.startsWith('Bearer ')) {
+      const acessToken = acessRawToken.substring(7, acessRawToken.length);
+      const key = await this._jwksClient.getSigningKey();
+      const publicKey = key.getPublicKey();
+      // this._debugger.log('canActivate get public key result: ', publicKey);
+
+      try {
+        const result = this.jwtService.verify(acessToken, { publicKey, complete: false });
+        this._debugger.log('canActivate token verification result: ', result)
+        const { tid: tenantId, aid: appId, sub: tenantUserId, scope, role, plan } = result;
+
+        if (appId) {
+          // Built in auth endpoints (REST) is granted by default
+          if (!parsedRequest.graphql && this._authPaths.includes(parsedRequest.resource)) {
+            this._debugger.log('canActivate request is part of whitelisted auth paths. Granting');
+            const anonymousUser = await this.authGuardService.buildAnonymousUser(tenantId)
+            AuthGuard._setAuthDataForRequest(parsedRequest, anonymousUser, appId);
+
+            return true;
+          }
+
+          this._debugger.log(`canActivate parsedRequest resource`, parsedRequest.resource);
+
+          const resource = parsedRequest.resource.split(' ').join('');
+          const authResponse = await this.authGuardService.isAuthorized(
+            tenantUserId,
+            tenantId,
+            resource,
+            scope,
+            role,
+            plan
+          );
+
+          this._debugger.log(`canActivate isAuthorized`, authResponse);
+          AuthGuard._setAuthDataForRequest(parsedRequest, authResponse.user, appId);
+
+          // Decide if the call is not granted and return
+          if (!authResponse.granted) {
+            this._debugger.log(`canActivate return false`);
+            return false;
+          }
+
+          // If granted, check tenant plan before returning
+          if (NebulrAuthService.isAnonymousUser(authResponse.user)) {
+            this._debugger.log(`canActivate no need to do plan restriction since this is a ANONYMOUS user`);
+            return true;
+          } else {
+            const hasRequiredPlan = this.authGuardService.hasRequiredPlan(authResponse.user.tenant.plan, resource);
+            this._debugger.log(`canActivate hasRequiredPlan`, hasRequiredPlan);
+
+            return hasRequiredPlan;
+          }
+
+        } else {
+          return false;
+        }
+      } catch (error) {
+        console.error(error);
+        throw error;
+      }
     } else {
-      const hasRequiredPlan = this.authGuardService.hasRequiredPlan(authResponse.user.tenant.plan, parsedRequest.resource);
```

Comment:
Where did the old way of using legacy authTokens go?

Reply:
restored